interfaces/apparmor/spec.go

```diff
@@ -146,6 +182,76 @@ func (spec *Specification) AddSnippet(snippet string) {
 	}
 }
 
+// RegisterSnippetKey adds a key to the list of valid keys
+func (spec *Specification) RegisterSnippetKey(key string) {
```

Comment:
I'm not sure how we plan to use this API?  we cannot re-register the same key, but there is no central place to call it from that I can easily think of

Reply:
my idea is to have a global list in this package of keys, new keys are created with something like:

var desktopFilesAASnippetKey = apparmor.NewSnippetKey("...")

this is defined in one of the builtin interfaces files are reused across them as needed.

there would be a helper appamor.AllSnippetKeys() []strings  that is mostly meant for use in tests to double check that the snippets defined across interfaces are the one we expect. There might some other introspection that might be useful for tests, not sure

Reply:
Ok, I did the change. Now the keys should be registered as top variables, and registering the same key twice will panic during startup, which will be detected when running any test.

Reply:
And another change to allow to register them in the init function if preferred.

Reply:
Of course, it's possible to only use top variables, but in that case the "id" would be unnecessary. What do you think?